### PR TITLE
Support Julia 1.0 and later only

### PR DESCRIPTION
**Pull request by omus**
Mainly I just wanted to add a Project.toml to the package.

### CODE REVIEW

File: src/Suppressor.jl [hunk 1]
```diff
@@ -1,14 +1,11 @@
 __precompile__()
 
 module Suppressor
-using Compat
 
 export @suppress, @suppress_out, @suppress_err
 export @capture_out, @capture_err
 export @color_output
 
-haslogging() = isdefined(Base, :CoreLogging)
```
Comment (omus) [inline src/Suppressor.jl:10]: Julia 1+ always has `Base.CoreLogging`. The remainder of these changes are me removing the use of this conditional and un-indenting code.

File: Project.toml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name = "Suppressor"
+uuid = "fd094767-a336-5f1f-9728-57cf17d0bbfb"
+version = "0.2.0"
```
Comment (omus) [inline Project.toml:3]: May be worth just jumping to `1.0.0` as this package hasn't seen many changes lately and seems rather stable.

File: .travis.yml [hunk 3]
```diff
@@ -4,9 +4,8 @@ os:
   - linux
   - osx
 julia:
-  - 0.6
-  - 0.7
   - 1.0
+  - 1.3
```
Comment (ssfrr) [inline .travis.yml]: Maybe best to test on versions `1.0` and `1`. The latter will test on the latest release of the `1.x` series.

File: appveyor.yml [hunk 4]
```diff
@@ -1,8 +1,7 @@
 environment:
   matrix:
-  - julia_version: 0.6
-  - julia_version: 0.7
   - julia_version: 1
+  - julia_version: 1.3
```
Comment (ssfrr) [inline appveyor.yml]: This setup will end up testing on 1.3 twice - as in the travis config probably best to do `1.0` and `1`.

**Review by ssfrr — CHANGES_REQUESTED** [review]
Some minor tweaks to the test config requested. Also it looks like there needs to be a test dependency added on `Test` to `Project.toml` so that CI can run.

File: .travis.yml (hunk 3 above)
Comment (omus) [reply .travis.yml]: I'll make the change

File: appveyor.yml (hunk 4 above)
Comment (omus) [reply appveyor.yml]: If you want I can switch Windows testing to be on Travis CI

**Review by iamed2 — APPROVED** [review]

File: Project.toml (hunk 2 above)
Comment (mcabbott) [reply Project.toml:3]: The registered version is `0.1.1`. 

If I'm reading correctly, this PR doesn't change much except dropping Julia 0.6. In which case, please tag it as `0.1.2`, not `0.2`. Even if you also tag it as `1.0` the day after. 

The reason I suggest this is that it currently bounds Compat.jl to <=2, here:
https://github.com/JuliaRegistries/General/blob/master/S/Suppressor/Compat.toml
If the end of any chain of old packages depends on Suppressor@0.1, and I need Compat@3 for some newer package, then these can't be installed together.

Comment (omus) [reply Project.toml:3]: > please tag it as 0.1.2

Due to changing the minimum required version of Julia this has to be at least a minor release. This is a requirement when registering packages.

Comment (mcabbott) [reply Project.toml:3]: Are you sure? I could be misinformed, but I thought e.g. this should be read as dropping Julia 0.6 on v0.3.1:
https://github.com/JuliaRegistries/General/blob/master/A/AxisArrays/Compat.toml

**Comment by JackDunnNZ** [comment]
Any chance this can be merged? As mentioned by @mcabbott (https://github.com/JuliaIO/Suppressor.jl/pull/33#discussion_r395902376) it is currently incompatible with Compat v3

File: Project.toml (hunk 2 above)
Comment (omus) [reply Project.toml:3]: I'm very confident: https://github.com/JuliaRegistries/RegistryCI.jl/blob/6b4ba393274dc8b3d5472751dc644d792697a956/src/AutoMerge/guidelines.jl#L79

It's actually hard to see this in General PRs as the comments are removed once the block has been resolved.

**Comment by omus** [comment]
I'm going to ahead and merge this. I was waiting for @ssfrr to approve.

**Comment by JackDunnNZ** [comment]
Thanks so much for the quick response!

**Pull request merged**